File: Api/src/Api/ModelsExports/Utilisateurs/UtilisateurExport.cs
using System.Text.Json.Serialization;

namespace Api.ModelsExports.Utilisateurs;

public sealed record UtilisateurExport
{
    public required string Nom { get; init; }

    public required string Prenom { get; init; }

    public required string Mail { get; init; }

    public required bool EstAdmin { get; init; }

    public string? Telephone { get; init; }
}

[JsonSerializable(typeof(UtilisateurExport[]))]
[JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase)]
public partial class UtilisateurExportContext: JsonSerializerContext { }

File: front/src/models/exports/UtilisateurExport.ts
export type UtilisateurExport =
{
    nom: string,
    prenom: string,
    mail: string,
    telephone?: string,
    estAdmin: boolean,
    mdp: string
}

export type UtilisateurModifierExport = Omit<UtilisateurExport, "mdp">;